### PR TITLE
PLAT-7525: Fix scenario when server would return a new recordedEntryI…

### PR DESCRIPTION
**Pull request by gadyaari**
…d and recordingManager would ignore it.

### CODE REVIEW

File: lib/recording/RecordingManager.js [hunk 1]
```diff
@@ -291,53 +304,59 @@ class RecordingManager {
...
-                logger.warn("[%s-%s][stopRecording] Entry duration hasn't changed, do nothing. start [%s] current[%s], totalAddedChunks [%s]", entryId, recordedEntryId, recordingEntrySession.getDuration(), recordingEntrySession.startDuration , totalAddedChunks)
-                delete this.recordingList[entryId];
-                return qio.write(doneFilePath, 'done')
+    stopRecording(entryId) {
+        if (this.recordingList[entryId]) {
+            let recordingEntrySession = this.recordingList[entryId];
+            let doneFilePath = path.join(recordingEntrySession.recordingSessionPath, 'done')
+            if (!recordingEntrySession) {
+                logger.warn("[%s][stopRecording] Can't stop recording, entry doesn't exist on recording list", entryId);
+                return
```
Comment (lilachmaliniak) [inline lib/recording/RecordingManager.js]: return promise

File: lib/recording/RecordingManager.js [hunk 2]
```diff
@@ -43,119 +44,136 @@ class RecordingManager {
...
                                 }
                             }
                         })
                     })
             }
-            const getRecordingEntrySession = ()=>{
-                let recordingEntrySession = null
+
+            function getRecordingEntrySession(entryObj) {
                 let entryId = entryObj.entryId;
```
Comment (kalturaguy) [inline lib/recording/RecordingManager.js]: you move this to be a function, so every this inside is a different scope... so proably everything is broken...

File: lib/recording/RecordingManager.js [hunk 3]
```diff
@@ -291,53 +304,59 @@ class RecordingManager {
...
-            let totalAddedChunks = _.reduce(recordingEntrySession.flavorsChunksCounter,(memo,flavorChunks)=>{
-                return memo+ flavorChunks
-            }, 0)
-            //todo should also print warning if # of chunks is not equall between all flavors!
-            let recordingSessionDuration = recordingEntrySession.getDuration() - recordingEntrySession.startDuration
-            if (! recordingSessionDuration > 0 ){
-                logger.warn("[%s-%s][stopRecording] Entry duration hasn't changed, do nothing. start [%s] current[%s], totalAddedChunks [%s]", entryId, recordedEntryId, recordingEntrySession.getDuration(), recordingEntrySession.startDuration , totalAddedChunks)
-                delete this.recordingList[entryId];
-                return qio.write(doneFilePath, 'done')
+    stopRecording(entryId) {
```
Comment (lilachmaliniak) [inline lib/recording/RecordingManager.js]: General comment:
stopRecording return promise that caller function doesn't handle. Please update code and test.

File: lib/recording/RecordingManager.js [hunk 4]
```diff
@@ -43,119 +44,136 @@ class RecordingManager {
...
+                    });
+                })
+                .catch((err)=> {
+                    // In case of error, pass message on and print to log
+                    return Q.reject(err);
+                });
         }).catch((err)=>{
-            logger.error("[%s-%s] Error while try to start recording : %s, delete object", entryObj.entryId, entryObj.recordedEntryId, ErrorUtils.error2string(err))
+            logger.error(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Error while try to start recording : ${ErrorUtils.error2string(err)}, delete object`);
             delete this.recordingList[entryObj.entryId]
```
Comment (lilachmaliniak) [inline lib/recording/RecordingManager.js]: check if there is item in recording list before delete. Please test this flow.

File: lib/recording/RecordingManager.js [hunk 5]
```diff
@@ -43,119 +44,136 @@ class RecordingManager {
...
 
-        setInterval(()=>{
+        setInterval(()=> {
             this.handleRecordingEntries()
-        }, recordingTimeInterval, this)
+        }, recordingTimeInterval, this);
     }
 
     startRecording(entryObj, flavorsObjArray){
         return Q.resolve().then(()=> {
```
Comment (lilachmaliniak) [inline lib/recording/RecordingManager.js]: General comment: if startRecording returned promise is rejected live entry will fail to start. I don't think we want that but is for product to decide. Please consult with Guy & Yaron and test the reject flow as well.

File: lib/recording/RecordingManager.js (hunk 2 above)
Comment (gadyaari) [inline lib/recording/RecordingManager.js]: I pass the function entryObj as a parameter, which is everything needed for it to run.
Also, tested it and nothing got broken. If still want me to change it be let me know

File: lib/recording/RecordingManager.js (hunk 4 above)
Comment (gadyaari) [inline lib/recording/RecordingManager.js]: If property doesn't exist in object delete has no effect and returns, so don't see the need to add another checkup here.
You can see the following: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

File: lib/recording/RecordingManager.js (hunk 3 above)
Comment (gadyaari) [inline lib/recording/RecordingManager.js]: Don't really understand what do you mean

File: lib/recording/RecordingManager.js (hunk 1 above)
Comment (gadyaari) [inline lib/recording/RecordingManager.js]: done.

File: lib/recording/RecordingManager.js (hunk 5 above)
Comment (gadyaari) [inline lib/recording/RecordingManager.js]: Good point. Will do

File: lib/recording/RecordingManager.js [hunk 6]
```diff
@@ -39,123 +40,141 @@ class RecordingManager {
...
                         })
                     })
             }
-            const getRecordingEntrySession = ()=>{
-                let recordingEntrySession = null
+
+            function getRecordingEntrySession(entryObj) {
                 let entryId = entryObj.entryId;
-                if (!("recordedEntryId" in entryObj && entryObj.recordedEntryId != undefined)){
+                if (!("recordedEntryId" in entryObj && entryObj.recordedEntryId != undefined)) {
```
Comment (kalturaguy) [inline lib/recording/RecordingManager.js]: replace entryObj.recordedEntryId != undefined with entryObj.recordedEntryId

File: lib/recording/RecordingEntrySession.js [hunk 7]
```diff
@@ -121,18 +121,18 @@ class RecordingEntrySession {
...
-            this.logger.info("New last update duration %s", duration)
-            return backendClient.updateEntryDuration(this.entryId, duration).then(()=> {
-                this.logger.info("Successfully update liveEntry %s with duration %s", this.entryId, duration)
-            })
+        let duration = this.getDuration();
+        if (force || duration > this.lastUpdateDuration + recordingUpdateEntryTimeIntervalMs) {
+            this.lastUpdateDuration = duration;
+            this.logger.info(`[${this.entryId}] New last update duration ${duration}`);
+            return backendClient.updateEntryDuration(this.entryId, duration).then((result)=> {
+                    this.logger.info(`[${this.entryId}] Successfully updated liveEntry with duration ${duration}`)
```
Comment (lilachmaliniak) [inline lib/recording/RecordingEntrySession.js]: Gad, if you return result then also log it.

**Pull request merged**